### DISCUSSION TITLE
Directory import is not supported resolving ES modules when importing `react-icons`

### DISCUSSION THREAD

**Discussion by offchan42** [remix-run/remix#7451] [Q&A] [answered]
I got ['Directory import is not supported resolving ES modules'](https://stackoverflow.com/questions/64453859/directory-import-is-not-supported-resolving-es-modules-with-node-js) error when I try to import [`react-icons` package](https://github.com/react-icons/react-icons) the way `react-icons` doc suggests me.

```jsx
// /app/routes/test-icons.tsx
import { FaLock } from 'react-icons/fa' // this line produces error (it's what react-icons doc suggests)
// import { FaLock } from 'react-icons/fa/index.js' // this line doesn't produce error


export default function TestIconsRoute() {
  return <FaLock />
}
```

1. Is this error expected and appending `/index.js` the correct workaround?
2. Which code line in Remix activates ES modules and causes me to need to append `/index.js`? 
3. Maybe this is a bit unrelated to Remix, which topic can I learn more about to understand the cause of this error better?

Context: I am using Remix v2.0.0 and I'm a bit new to web dev.

**Comment by abdulkader** [comment]
This started breaking from v2.0.0. Here is reproducible repo with just basic setup https://github.com/abdulkader/remix-admin-template

**Comment by abdulkader** [comment]
@off99555 I have resolved the issue by adding the option `serverDependenciesToBundle: [/^react-icons/],` to the `remix.config.js` file. You can see same is working on above repo.

**Reply by offchan42** [reply]
Is this the correct way to fix the issue or is this a workaround?

**Reply by aprilmintacpineda** [reply]
The issue is still happening today, this worked for me.

**Comment by hanqitech** [comment]
https://github.com/mui/material-ui/tree/master/examples/material-ui-remix-ts
I solved the problem according to this ,hope this could help you .

**Comment by RabeeAbuBaker** [comment, marked as answer]
This has been [fixed](https://github.com/react-icons/react-icons/pull/880) by [react-icons](https://github.com/react-icons/react-icons) ... just upgrade to v5

https://github.com/react-icons/react-icons/issues/717
